src/main/java/co/cask/hydrator/plugin/DynamoDBRecordWriter.java

```diff
@@ -0,0 +1,119 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.hydrator.plugin;
+
+import co.cask.hydrator.plugin.utils.DynamoDBConstants;
+import co.cask.hydrator.plugin.utils.DynamoDBUtil;
+import com.amazonaws.services.dynamodbv2.document.BatchWriteItemOutcome;
+import com.amazonaws.services.dynamodbv2.document.DynamoDB;
+import com.amazonaws.services.dynamodbv2.document.Item;
+import com.amazonaws.services.dynamodbv2.document.TableWriteItems;
+import com.amazonaws.services.dynamodbv2.document.internal.InternalUtils;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ProvisionedThroughputExceededException;
+import com.amazonaws.services.dynamodbv2.model.WriteRequest;
+import org.apache.hadoop.io.NullWritable;
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Record Writer for DynamoDB Sink.
+ */
+public class DynamoDBRecordWriter extends RecordWriter<NullWritable, Item> {
+  private static final Logger LOG = LoggerFactory.getLogger(DynamoDBRecordWriter.class);
+  int batchSizeBytes = 0;
+  private DynamoDB dynamoDB;
+  private String tableName;
+  private List<Item> itemList = new ArrayList<>();
+
+  public DynamoDBRecordWriter(TaskAttemptContext context) throws IOException {
+    dynamoDB = new DynamoDB(DynamoDBUtil.getDynamoDBClient(context.getConfiguration()));
+    tableName = context.getConfiguration().get(DynamoDBConstants.OUTPUT_TABLE_NAME);
+  }
+
+  @Override
+  public void write(NullWritable key, Item item) throws IOException, InterruptedException {
+    if (item == null) {
+      throw new RuntimeException("Null record encountered. At least the key columns must be specified.");
+    }
+    Map<String, AttributeValue> attributes = InternalUtils.toAttributeValues(item);
+    int itemSizeBytes = DynamoDBUtil.getItemSizeBytes(attributes);
+    if (itemSizeBytes > DynamoDBConstants.MAX_ALLOWABLE_BYTE_SIZE) {
+      throw new RuntimeException(String.format("Cannot pass items with size greater than %s bytes",
+                                               DynamoDBConstants.MAX_ALLOWABLE_BYTE_SIZE));
+    }
+    batchSizeBytes += itemSizeBytes;
+
+    itemList.add(item);
+
+    if (itemList.size() == DynamoDBConstants.MAX_WRITE_BATCH_SIZE
+      || batchSizeBytes >= DynamoDBConstants.MAX_BATCH_SIZE_BYTES) {
+      putBatch(itemList);
+      itemList.clear();
+      batchSizeBytes = 0;
+    }
+  }
+
+  private void putBatch(List<Item> items) {
+    TableWriteItems tableWriteItems = new TableWriteItems(tableName).withItemsToPut(items);
+    for (int i = 0; i < 3; i++) {
```

Comment:
It would be good to export 3 to a Maximum_retry_variable. makes it more readable and easy to extend.